mapstructure.go

```diff
@@ -588,9 +588,13 @@ func (d *Decoder) decodeSlice(name string, data interface{}, val reflect.Value)
 		// Check input type
 		if dataValKind != reflect.Array && dataValKind != reflect.Slice {
 			// Accept empty map instead of array/slice in weakly typed mode
-			if d.config.WeaklyTypedInput && dataVal.Kind() == reflect.Map && dataVal.Len() == 0 {
+			if d.config.WeaklyTypedInput && dataValKind == reflect.Map && dataVal.Len() == 0 {
 				val.Set(reflect.MakeSlice(sliceType, 0, 0))
 				return nil
+			} else if d.config.WeaklyTypedInput && dataValKind == reflect.String {
+				slc := []string{dataVal.String()}
+				val.Set(reflect.ValueOf(slc))
```

Comment:
This might not work because the type may not match up (if the slice type isn't string), I think. I think you have to make a slice of the sliceType and try to weak decode into it. But I'm not 100% sure. I'll take a look too.